### ISSUE TITLE
Error in function (type, msg, asError = TRUE)  : Empty reply from server 

### ISSUE THREAD

**Issue by LeuteMatthias** [fcocquemas/rdatastream#8]
Another issue that I discovered is an empty reply from server which also ends my function. Unfortunately it occurred 5 hours in the loop so I'll keep on testing. However, I encountered the error yesterday for another firm, too.

```
6.
fun(structure(list(message = msg, call = sys.call()), class = c(typeName, 
    "GenericCurlError", "error", "condition"))) 
5.
function (type, msg, asError = TRUE) 
{
    if (!is.character(type)) {
        i = match(type, CURLcodeValues) ... 
4.
curlPerform(curl = curl, .opts = opts, .encoding = .encoding) 
3.
getURL(wsdl, httpheader = c(Accept = "text/xml", Accept = "multipart/*", 
    `Content-Type` = "application/soap+xml; charset=utf-8"), 
    postfields = xmlRequest, verbose = FALSE) 
2.
ds(user, securities = Firms[f], fields = Variables[v], fromDate = "2002-12-31", 
    toDate = "2015-12-31", period = "Y") 
1.
DataStream(Firms_1501_2000, Variables) 
```

![screen shot 2017-05-25 at 10 14 11 am](https://cloud.githubusercontent.com/assets/25764046/26441986/fac4b97a-4132-11e7-82d4-f9154ce2bd70.png)

Best Regards,
Matthias

**Comment by fcocquemas** [comment]
Hello Matthias,

Unless you can reproduce this error for this particular request, this sounds to me like a temporary server error on their side, not a library error. 

Check out [section 4.1.2](http://dataworks.thomson.com/Dataworks/Enterprise/1.0/documentation/user%20guide.pdf) of the Dataworks Enterprise coding guideline to make sure you're not overwhelming the server. Are you using `parallel` or a simple loop?

I'm looking into your other issue now. Thanks for reporting this!

Best,
François

**Comment by LeuteMatthias** [comment]
Hi François,
I was able to reproduce the error multiple times however, it appears always for different firms (at different times in the loop). So it could very well be that it is a temporary server error on their side.
For now I solved the issue by having only 200 firms at once (batch sizes of 500 firms before) in the loop and when the error occurs I simply start the download over again.

Best,
Matt
